tests/integration_tests/functional/test_drives.py

```diff
@@ -329,6 +329,107 @@ def test_patch_drive(test_microvm_with_ssh, network_config):
     assert stdout.readline().strip() == size_bytes_str
 
 
+def check_iops_limit(ssh_connection, block_size, count, min_time, max_time):
+    """Verify if the rate limiter throttles block iops using dd."""
+    byte_count = block_size * count
+    dd = "dd if=/dev/vdb of=/dev/null ibs={} count={} iflag=direct".format(
+        block_size, count)
+    print("Running cmd {}".format(dd))
+    # Check read iops.
+    exit_code, _, stderr = ssh_connection.execute_command(dd)
+    assert exit_code == 0
+
+    # "dd" writes to stderr by design. We drop first lines
+    stderr.readline().strip()
+    stderr.readline().strip()
+    dd_result = stderr.readline().strip()
+
+    # Interesting output looks like this:
+    # 4194304 bytes (4.2 MB, 4.0 MiB) copied, 0.0528524 s, 79.4 MB/s
+    tokens = dd_result.split()
+    print(tokens)
+    # Check total read bytes.
+    assert int(tokens[0]) == byte_count
+    # Check duration.
+    assert float(tokens[7]) > min_time
+    assert float(tokens[7]) < max_time
+
+
+def test_patch_drive_limiter(test_microvm_with_ssh, network_config):
+    """Test replacing the drive rate-limiter after guest boot works."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.jailer.daemonize = False
+    test_microvm.spawn()
+    # Set up the microVM with 2 vCPUs, 512 MiB of RAM, 1 network iface, a root
+    # file system with the rw permission, and a scratch drive.
+    test_microvm.basic_config(vcpu_count=2,
+                              mem_size_mib=512,
+                              boot_args='console=ttyS0 reboot=k panic=1')
+
+    _tap, _, _ = test_microvm.ssh_network_config(network_config, '1')
+
+    fs1 = drive_tools.FilesystemFile(
+        os.path.join(test_microvm.fsfiles, 'scratch'),
+        size=512
+    )
+    response = test_microvm.drive.put(
+        drive_id='scratch',
+        path_on_host=test_microvm.create_jailed_resource(fs1.path),
+        is_root_device=False,
+        is_read_only=False,
+        rate_limiter={
+            'bandwidth': {
+                'size': 1024*1024*10,
+                'refill_time': 100
+            },
+            'ops': {
+                'size': 100,
+                'refill_time': 100
+            }
+        }
+    )
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    test_microvm.start()
+    ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
+
+    check_iops_limit(ssh_connection, 512, 1000, 0.85, 1.25)
```

Comment:
I might lack the context for rate-limiting, but I think that comments in regards to where these numbers come from would help

Reply:
I think it is clear if you look at the fn prototype and doc line:
```
def check_iops_limit(ssh_connection, block_size, count, min_time, max_time):
    """Verify if the rate limiter throttles block iops using dd."""
```

Based on the limiter confioguration we expect to do that much IO (count * block_size ) in a specific interval of time while allowing some variance.

Reply:
Added comments.